management-api-agent-common/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java

```diff
@@ -740,39 +737,46 @@ public void clearSnapshots(
   @Rpc(name = "repair")
   public String repair(
       @RpcParam(name = "keyspaceName") String keyspace,
-      @RpcParam(name = "tables") Optional<List<String>> tables,
+      @RpcParam(name = "tables") List<String> tables,
       @RpcParam(name = "full") Boolean full,
       @RpcParam(name = "notifications") boolean notifications,
-      @RpcParam(name = "repairParallelism") Optional<RepairParallelism> repairParallelism,
-      @RpcParam(name = "datacenters") Optional<Collection<String>> datacenters,
-      @RpcParam(name = "associatedTokens") Optional<List<RingRange>> associatedTokens,
-      @RpcParam(name = "repairThreadCount") Optional<Integer> repairThreadCount)
+      @RpcParam(name = "repairParallelism") String repairParallelism,
+      @RpcParam(name = "datacenters") List<String> datacenters,
+      @RpcParam(name = "associatedTokens") String ringRangeString,
+      @RpcParam(name = "repairThreadCount") int repairThreadCount)
```

Comment:
Question: Slightly concerned that we can't distinguish a zero from a null here? Would it be better to use an Integer? Or can we not serialise that?

Reply:
For thread count, we could use an Integer object, but then it adds a null-check in at least the NodeOpsProvider. With it being a primitive, the caller will fail if an `int` isn't sent over the wire. And the logic in NodeOpsProvider is:
```java
RepairOption.JOB_THREADS_KEY, Integer.toString(Integer.max(1, repairThreadCount))
```
so anything less than `1` becomes `1`, so the logic would simply set `0` and `null` to a `1` anyway.

Reply:
The thing is, I think we want to leave this unset if it is null, so indeed there should be a null check and we shouldn't add it to the options map.

Reply:
OK, I will update it then to do it like that.